### PR TITLE
IMU addition proper

### PR DESCRIPTION
**Pull request by Kredcool**
adding proper IMU commands such as generate and ext.

### CODE REVIEW

File: rio/robot.py [hunk 1]
```diff
@@ -47,6 +47,7 @@ def teleopInit(self) -> None:
         # this line or comment it out.
         if self.autonomousCommand is not None:
             self.autonomousCommand.cancel()
+        self.container.robotDrive.imu.getRawGyro()
```
Comment (KenwoodFox) [inline rio/robot.py]: What does this line do?

File: rio/subsystems/drivesubsystem.py [hunk 2]
```diff
@@ -159,8 +159,8 @@ def zeroHeading(self):
         """
         Zeroes the heading of the robot.
         """
-        # This is most likey the wrong was to do
-        # this but I can't find the reset command
+        # TODO find the proper reset command
+        # because this is surely wrong
         self.imu.configMountPose(
             self.imuConst["yaw"],
             self.imuConst["pitch"],
```
Comment (KenwoodFox) [inline rio/subsystems/drivesubsystem.py]: This does not reset anything it just defines where the imu is on the robot.

File: rio/subsystems/drivesubsystem.py [hunk 3]
```diff
@@ -172,11 +172,13 @@ def getHeading(self):
         Returns the heading of the robot.
         :returns: the robot's heading in degrees, from 180 to 180
         """
-        return geometry.Rotation2d.fromDegrees(self.imu.getYaw())
+        return geometry.Rotation2d.fromDegrees(self.imu.getYawPitchRoll())
```
Comment (KenwoodFox) [inline rio/subsystems/drivesubsystem.py]: this is incorrect, it should be `self.imu.getYaw()`

File: rio/subsystems/drivesubsystem.py [hunk 4]
```diff
@@ -172,11 +172,13 @@ def getHeading(self):
...
 
     def getTurnRate(self):
         """
         Returns the turn rate of the robot.
         :returns: The turn rate of the robot, in degrees per second
         """
-        return self.imu.GetRawGyro()
+        return (
+            self.imu.getRawGyro()
+        )  # I think this works, but take it with a grain of salt and the knowlege it could I could be dumb -- Keegan 2023 build season
```
Comment (KenwoodFox) [inline rio/subsystems/drivesubsystem.py]: Test it!

**Review by KenwoodFox — CHANGES_REQUESTED** [review]

File: rio/robot.py (hunk 1 above)
Comment (Kredcool) [reply rio/robot.py]: I meant to delete that, it was for testing purposes, just forgot to save the file.

**Comment by KenwoodFox** [comment]
Lets order a https://www.andymark.com/products/navx2-mxp-robotics-navigation-sensor and prepare to makeup a drive-alike for drivers to practice on. So far i have not been impressed by the input and control from this robot.

**Comment by KenwoodFox** [comment]
You should test if this actually works on the bot, the diff lock and gyro turn commands.

**Review by KenwoodFox — APPROVED** [review]
This is all cleaned up now, this should be tested just a little bit. But this pr has obviously been growing quite large. I might suggest merging it, and making another dedicated branch to tune the navx.

you have a navx in one of the box rios, you should plug that in and see if you can at least get some basic nav data from it before merging.

**Pull request merged**